app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,62 @@
+class ListingsController < ApplicationController
+    before_action :authenticate_user!
+
+  def index
+    @listing = Listing.new
+    @listingList = Listing.where(user_id: current_user.id).paginate(page: params[:page], per_page: 3)
+  end
+
+  def show
+    @listing = Listing.find(params[:id])
+    @hostDetails = User.find(@listing.user_id)
+  end
+
+  def create
+    @listing = Listing.new(listing_params)
+    if @listing .save
+      flash[:notice] = 'Listing added successfully'
+      redirect_to listings_url
+    else
+      flash[:alert] = 'Failed to add Listing'
+      redirect_to listings_url
+    end
+  end
+
+  def update
+    @listing = Listing.find(params[:id])
+    if @listing.update_attributes(listing_params)
+      flash[:notice] = 'Updated Listing Details!'
+      redirect_to @listing 
+    else
+      flash[:alert] = 'Update Failed!'
+      redirect_to @listing 
```

Comment:
Put `redirect_to @listing ` after the `if-else-end` so that it is only called once.